app/helpers/spotlight/main_app_helpers.rb

```diff
@@ -27,4 +27,18 @@ def field_enabled? field, *args
     field.enabled && field.send((:show if controller.is_a? Blacklight::Catalog and ["edit", "show"].include?(action_name)) || document_index_view_type)
   end
 
+  def link_back_to_catalog(opts={:label=>nil})
+    opts[:route_set] ||= spotlight if current_exhibit
+    super
+  end
+
+  def render_save_search
```

Comment:
I think with https://github.com/projectblacklight/blacklight/commit/6e5583e525c38ebf215663ea9923c1bc29a9d909, we should be able to do this better (and, in fact, the override we're doing right now needs to be updated too..)